### PR TITLE
Remove Azure Storage

### PR DESCRIPTION
**Pull request by ckcollab**
Make Codalab able to switch between Azure and S3 -- this switch **must** be made at the beginning or all old files removed.

## Key generation
- [x] `competition_views.py` and `CompetitionCreationSasApi`

## Competition creation
- [x] Competition uploading
- [x] Competition upload progress...
- [x] Make competition bundles private, requiring key to download
- [x] Put competitions, ref data, etc. in private storage

## Submissions
- [x] `getBundle()` usage switches to S3 version
- [x] Submission uploading
- [x] `_upload()` usages switch to S3 versions

## Settings
- [x] `deploy/__init__.py` line 325 `"    DEFAULT_FILE_STORAGE = 'codalab.azure_storage.AzureStorage'",`
- [x] `local.py`
- [ ] `local_sample.py`

### CODE REVIEW

**Comment by lotusgraham** [comment]
These are some screenshots of the part of the user interface tweaked by last commit. It's responsive, although i doubt people will be uploading competitions from their phones. ;-)

screen shot 2017-02-23 at 2 04 43 pm

screen shot 2017-02-23 at 2 05 00 pm

**Comment by ckcollab** [comment]
Nice, thats looking good! can you align the "No File Chosen" text?

**Comment by lotusgraham** [comment]
a screenshot up to this point. This is after a competition is uploaded.
screen shot 2017-02-24 at 4 54 44 pm

**Comment by ckcollab** [comment]
@lotusgraham Could you hide the filename as well? I don't think the other version shows that -- sorry for all the nitpicky stuff =\

File: codalab/apps/web/static/less/my.less [hunk 1]
```diff
@@ -24,3 +24,49 @@
...
+  }
+
+  .btn-primary{
+    width: 50%;
+    min-width: 200px;
+    margin: 25px 25%;
+  }
+
+  .s3direct {
+      margin: 5 auto;
```
Comment (ckcollab) [inline codalab/apps/web/static/less/my.less]: @lotusgraham Do you need to add px here or something?

File: codalab/apps/web/templates/web/competitions/upload_s3_competition.html [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            </div>
+
+            <form id="s3_upload_form" action="" method="post">
+                {% csrf_token %}
+                <div class="form-group">
+                  <button type="button" id="submit-button" class="btn btn-primary">Upload Competition Bundle</button>
+                </div>
+                <div class="row">
+                  <div class="form-group center-block" style="width:50%; height: 10px;">
+                    {{ form }}
```
Comment (ckcollab) [inline codalab/apps/web/templates/web/competitions/upload_s3_competition.html]: I'm still seeing the file name after successful upload?

Comment (lotusgraham) [inline codalab/apps/web/templates/web/competitions/upload_s3_competition.html]: This is an informative bit of User Interface for people to see the filename of what they've uploaded.

**Pull request merged**